apps/rush-lib/src/cli/actions/ListAction.ts

```diff
@@ -25,43 +28,75 @@ export class ListAction extends BaseRushAction {
     this._version = this.defineFlagParameter({
       parameterLongName: '--version',
       parameterShortName: '-v',
-      description: 'If this flag is specified, the project version will be ' +
+      description:
+        'If this flag is specified, the project version will be ' +
         'displayed in a column along with the package name.'
     });
 
     this._path = this.defineFlagParameter({
       parameterLongName: '--path',
       parameterShortName: '-p',
-      description: 'If this flag is specified, the project path will be ' +
+      description:
+        'If this flag is specified, the project path will be ' +
         'displayed in a column along with the package name.'
     });
 
     this._fullPath = this.defineFlagParameter({
       parameterLongName: '--full-path',
       parameterShortName: '-f',
-      description: 'If this flag is specified, the project full path will ' +
+      description:
+        'If this flag is specified, the project full path will ' +
         'be displayed in a column along with the package name.'
     });
+
+    this._jsonFlag = this.defineFlagParameter({
+      parameterLongName: '--json',
+      description: 'If specified, output will be in JSON format.'
+    });
   }
 
-  protected run(): Promise<void> {
+  protected async run(): Promise<void> {
     return Promise.resolve().then(() => {
       const allPackages: Map<string, RushConfigurationProject> = this.rushConfiguration.projectsByName;
-      if (this._version.value || this._path.value || this._fullPath.value) {
+      if (this._jsonFlag.value) {
+        this._printJson(allPackages);
+      } else if (this._version.value || this._path.value || this._fullPath.value) {
         this._printListTable(allPackages);
       } else {
         this._printList(allPackages);
       }
     });
   }
 
-  private _printList(allPackages: Map<string, RushConfigurationProject>): void {
+  private _printJson(
+    allPackages: Map<string, RushConfigurationProject>
+  ): void {
+    allPackages.forEach((_config: RushConfigurationProject, name: string) => {
+     const jsonString: string = JSON.stringify({
+       name: name,
+       version: _config.packageJson.version,
+       path: _config.projectRelativeFolder, fullPath: _config.projectFolder});
+     this._objects[this._count++] = JSON.parse(jsonString);
```

Comment:
Why are you serializing and then immediately parsing this object?

Reply:
I thought  we are going to return an array of JSON objects but it seems I got it misunderstood. Will remove the parsing part. :)